src/browser-installer/registry/index.ts

```diff
@@ -0,0 +1,215 @@
+import type { BrowserPlatform } from "@puppeteer/browsers";
+import { readJsonSync, outputJSONSync, existsSync } from "fs-extra";
+import path from "path";
+import {
+    getRegistryPath,
+    browserInstallerDebug,
+    Driver,
+    Browser,
+    getMilestone,
+    normalizeChromeVersion,
+    semverVersionsComparator,
+    type SupportedBrowser,
+    type SupportedDriver,
+    type DownloadProgressCallback,
+} from "../utils";
+import { getFirefoxBuildId } from "../firefox/utils";
+import logger from "../../utils/logger";
+import type { createBrowserDownloadProgressBar } from "./cli-progress-bar";
+
+type VersionToPathMap = Record<string, string | Promise<string>>;
+type BinaryName = Exclude<SupportedBrowser | SupportedDriver, SupportedBrowser & SupportedDriver>;
```

Comment:
That way if `SupportedBrowser` and `SupportedDriver` would have intersection, we wouldn't be able to use binary name neither as browser name, nor as driver name